File: session-1/node/led.js
var Gpio = require('onoff').Gpio;



function LED(pin){
	this.pin = parseInt(pin,10);
	this.gpio = new Gpio(pin,'out');
}


LED.prototype.turnOn = function(callback){
	if(callback){
		this.gpio.write(0, callback);
	}else{
		this.gpio.writeSync(0);
	}
}

LED.prototype.turnOff = function(callback){
	if(callback){
		this.gpio.write(1,callback);
	}else{
		this.gpio.writeSync(1);
	}
}

LED.prototype.cleanup = function(){
	this.gpio.unexport();
}

module.exports = LED;

File: session-2/python/led.py
import RPi.GPIO as GPIO
import time

class LED:
	"""Simple LED class"""

	def __init__(self, pin):
		self.pin = int(pin); #make sure it's a number.
		if(self.pin > 30):
			raise Exception("Cannot use pin higher than 30")
		GPIO.setmode(GPIO.BCM) #switch mode
		GPIO.setwarnings(False)
		GPIO.setup(self.pin,GPIO.OUT)

	def turnOn(self):
		GPIO.output(self.pin,False)

	def turnOff(self):
		GPIO.output(self.pin,True)

	def isOn(self):
		return False

	def wait(self,interval):
		time.sleep(interval)

	#allow cleaning up
	def cleanup(self):
		GPIO.cleanup();
